docs/design/persistent-volume-provisioning.md

```diff
@@ -0,0 +1,140 @@
+	Selector map[string]string `json:"selector"`
+
+	// Template is the description of a PersistentVolume to create new replicas from
+	Template *PersistentVolumeTemplateSpec `json:"template,omitempty"`
+}
+
+// PersistentVolumeControllerStatus represents the current status of a PVController
+type PersistentVolumeControllerStatus struct {
+	// BoundReplicas is the number of replicas of this volume that are currently bound to PersistentVolumeClaims
+	BoundReplicas int `json:"boundReplicas"`
+	UnboundReplicas int `json:"unboundReplicas"`
+}
+
+// PersistentVolumeTemplateSpec describes the persistent volume created by this controller
+type PersistentVolumeTemplateSpec struct {
+	// Metadata of the PVs created from this template.
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines a persistent volume.
+	Spec PersistentVolumeSpec `json:"spec,omitempty"`
+}
+
+```
+
+Dynamic pools of storage (e.g, GCE) will maintain `MinimumReplicas` of a `PersistentVolume` at a given time where each PVs status phase is ` Available`, up to `MaximumReplicas`.  Storage levels will increase by one until the minimum or maximum levels are reached.  Recycling a PV means the current number of available volumes might exceed the minimum replica count.  Plugins implementing the Deleter interface will delete PVs where the `PersistentVolumeReclaimPolicy` is set to Delete.  The default reclaim policy is Retain.
+
+Static pools of storage (e.g, NFS) will have ```MinimumReplicas``` == ```MaximumReplicas```.  Creation or deletion of the resource will not occur but the controller allows recycling of the volume through the plugin.
+
+#### PersistentVolumeController API
+
+| Action | HTTP Verb | Path | Description |
+| ---- | ---- | ---- | ---- |
+| CREATE | POST | /api/{version}/persistentvolumecontrollers/ | Create instance of PersistentVolumeController|
+| GET | GET | /api/{version}persistentvolumecontrollers/{name} | Get instance of PersistentVolumeController |
+| UPDATE | PUT | /api/{version}/persistentvolumecontrollers/{name} | Update instance of PersistentVolumeController |
+| DELETE | DELETE | /api/{version}/persistentvolumecontrollers/{name} | Delete instance of PersistentVolumeController |
+| LIST | GET | /api/{version}/persistentvolumecontrollers | List instances of PersistentVolumeController|
+| WATCH | GET | /api/{version}/watch/persistentvolumecontrollers | Watch for changes to a PersistentVolumeController|
+
+
+#### PersistentVolumeControllerManager
+
+* watch PersistentVolumes of a particular type as defined by the controller's label selector.
+** PVs matching the selector but made manually count as replicas if they match the label selector
+** PVs made manually or from earlier templates may have difference size capacities.  Delete largest first.
+* create a new volume ``` if unboundReplicas < minimumReplicas && totalReplicas <= maximumReplicas``` 
+* delete a volume ``` while totalReplicas > maximumReplicas, delete largest volume ```
```

Comment:
would you ever delete a bound replica? it seems you could get into that situation if someone manually creates a volume, outside of the controller, with a matching label.